### PR TITLE
Users should be directly able to navigate to the product detail page (#1yzcv99)

### PR DESCRIPTION
**Pull request by rathoreprashant**
### Related Issues
Improved: code handled fetchProducts by catch

Closes #

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vsf-capybara/blob/master/CONTRIBUTING.md)
- [ ] I have changed the target to hacktoberfest branch

### CODE REVIEW

File: src/store/modules/product/actions.ts [hunk 1]
```diff
@@ -83,10 +83,21 @@ const actions: ActionTree<ProductState, RootState> = {
...
-    commit(types.PRODUCT_CURRENT_UPDATED, { product: currentProduct ? currentProduct : payload.product })
+    const currentProduct = state.uploadProducts[payload]
+    if(currentProduct) {
+      commit(types.PRODUCT_CURRENT_UPDATED, { product: currentProduct })
+    } else {
+      await ProductService.fetchProducts({
+        // used sku as we are currently only using sku to search for the product
+        "filters": ['sku: ' + '*' + payload + '*', 'isVirtual: false'],
+      }).then((resp)=>{
+        commit(types.PRODUCT_CURRENT_UPDATED, { product: resp.data.response.docs[0] })
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts:96]: Handle the case when we have a response, but the status of the response is not 200, or the response contains error field, or the response has empty data.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/product/actions.ts [hunk 2]
```diff
@@ -83,10 +83,21 @@ const actions: ActionTree<ProductState, RootState> = {
...
+      }).catch((error)=>{
+        console.log(error);
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts:98]: Make it console.error

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/product/actions.ts (hunk 2 above)
Comment (rathoreprashant) [reply src/store/modules/product/actions.ts:98]: ok sir

Comment (rathoreprashant) [reply src/store/modules/product/actions.ts]: Improved sir

File: src/store/modules/product/actions.ts [hunk 3]
```diff
@@ -83,10 +83,24 @@ const actions: ActionTree<ProductState, RootState> = {
...
+    const currentProduct = state.uploadProducts[payload]
+    if(currentProduct) {
+      commit(types.PRODUCT_CURRENT_UPDATED, { product: currentProduct })
+    } else {
+    try {
+      const resp = await ProductService.fetchProducts({
+        // used sku as we are currently only using sku to search for the product
+        "filters": ['sku: ' + '*' + payload + '*', 'isVirtual: false'],
+      })
+      if (resp.status === 200 && resp.data.response?.numFound > 0 && !hasError(resp)) {
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts:97]:
```suggestion
      if (resp.status === 200 && resp.data?.response?.numFound > 0 && !hasError(resp)) {
```

File: src/store/modules/product/actions.ts [hunk 4]
```diff
@@ -83,10 +83,24 @@ const actions: ActionTree<ProductState, RootState> = {
...
+          commit(types.PRODUCT_CURRENT_UPDATED, { product: resp.data.response.docs[0] })
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts:98]: Improve indentation

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/product/actions.ts (hunk 4 above)
Comment (rathoreprashant) [reply src/store/modules/product/actions.ts:98]: ok sir

Comment (rathoreprashant) [reply src/store/modules/product/actions.ts:98]: Fixed Sir

File: src/store/modules/product/actions.ts (hunk 3 above)
Comment (rathoreprashant) [reply src/store/modules/product/actions.ts]: Updated sir

**Comment by disha1202** [comment]
Not needed, closing.

**Pull request closed without merging**